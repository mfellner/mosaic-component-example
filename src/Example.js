import React, { Component } from 'react'
import CSSModules from 'react-css-modules'
import styles from './styles.css'

class Example extends Component {
  render() {
    const level = this.props.level > 0 && this.props.level < 7 ? this.props.level : 1
    return <span styleName={`example example-${level}`}>
      {this.props.children}
    </span>
  }
}

Example.propTypes = {
  level: React.PropTypes.number
}

export default CSSModules(Example, styles, {allowMultiple: true})
