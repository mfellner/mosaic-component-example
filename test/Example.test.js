import React from 'react'
import renderer from 'react-test-renderer'
import Example from '../src/Example'

test('Example should add a CSS class by default.', () => {
  const component = renderer.create(
    <Example>Hello, world!</Example>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Example should add a CSS class with the given level.', () => {
  const component = renderer.create(
    <Example level={3}>Hello, world!</Example>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
