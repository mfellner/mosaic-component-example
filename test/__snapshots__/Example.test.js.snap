exports[`test Example should add a CSS class by default. 1`] = `
<span
  className="example example-1">
  Hello, world!
</span>
`;

exports[`test Example should add a CSS class with the given level. 1`] = `
<span
  className="example example-3">
  Hello, world!
</span>
`;
